packages/terra-date-picker/src/DateRange.jsx

```diff
@@ -0,0 +1,80 @@
+import React, { PropTypes } from 'react';
+import DatePicker from './DatePicker';
+
+const propTypes = {
+  /**
+   * A moment object to use as the default end date for a date range.
+   */
+  endDate: PropTypes.oneOfType([PropTypes.object]),
+  /**
+   * A moment object to use as the default start date for a date range.
+   */
+  startDate: PropTypes.oneOfType([PropTypes.object]),
+  /**
+   * A callback function to execute when a valid date is selected or entered.
+   */
+  onChange: PropTypes.func,
+};
+
+class DateRange extends React.Component {
+
+  constructor(props) {
+    super(props);
+    this.state = {
+      startDate: props.startDate,
+      endDate: props.endDate,
+    };
+    this.handleChangeStart = this.handleChangeStart.bind(this);
+    this.handleChangeEnd = this.handleChangeEnd.bind(this);
+  }
+
+  handleChange({ startDate, endDate }) {
+    let startDateForRange = startDate || this.state.startDate;
+    let endDateForRange = endDate || this.state.endDate;
+
+    if (startDateForRange.isAfter(endDateForRange)) {
+      const tempDate = startDateForRange;
+      startDateForRange = endDateForRange;
+      endDateForRange = tempDate;
+    }
+
+    this.setState({ startDate: startDateForRange, endDate: endDateForRange });
+
+    if (this.props.onChange) {
+      this.props.onChange(startDateForRange, endDateForRange);
+    }
+  }
+
+  handleChangeStart(startDate) {
+    this.handleChange({ startDate });
+  }
+
+  handleChangeEnd(endDate) {
+    this.handleChange({ endDate });
+  }
+
+  render() {
+    return (<div className="terra-DatePicker-range">
+      <DatePicker
+        {...this.props}
```

Comment:
For our other components, we have been putting customProps (`...this.props`) last. I think this was to allow customProps to be last and override attributes. Can someone else confirm this is correct?